### PR TITLE
Update Conformance chapter

### PR DESCRIPTION
**Pull request by bact**
From Bob's Word doc 2024-07-22 with few edits:

- Translate links to a preferred format, with text labels
- In Software Profile section, fix typo: "Conformance to the security profile" -> "Conformance to the Software Profile"
- Recover corrupted text in section 5.12 using the same section from existing [conformance.md](https://github.com/spdx/spdx-spec/blob/development/v3.0.1/docs/conformance.md) (from v2.2.1, v2.3)

### Notes

- Dataset profile is spelled out as "Data Set" in the Word doc
  - AI/Data Team Meeting 2024-07-24 suggests we should go with "Dataset" just as it is in the spec
- The URL for "SPDX Spec Git Repository Schema directory" https://github.com/spdx/spdx-spec/blob/master/schemas/spdx-schema.json is not exist, have to fix this
- See #987 for related discussion to subsection "5.12 Standard data format requirements"

### CODE REVIEW

**Review by bobmartin3000 — APPROVED** [review]
LGTM

**Review by kestewart — APPROVED** [review]
Thanks for pulling this together.

File: docs/conformance.md [hunk 1]
```diff
@@ -1,83 +1,299 @@
...
+  exchanged. Current supported formats include:
+
+  - **YAML 1.2**
+    see:
+    [YAML Ain’t Markup Language (YAML™) version 1.2](https://yaml.org/spec/1.2/)
+  - **JavaScript Object Notation** (JSON)
+    see:
+    [ECMA-404](https://ecma-international.org/publications-and-standards/standards/ecma-404/)
+    - The JSON Schema for SPDX can be found in the
+      [SPDX specification Git repository schemas directory](https://github.com/spdx/spdx-spec/blob/master/schemas/spdx-schema.json)
```
Comment (bact) [inline docs/conformance.md:266]: This URL is not exist

Comment (kestewart) [reply docs/conformance.md:266]: @bact - good catch.

Comment (kestewart) [reply docs/conformance.md:266]: TO Discuss:  should we merge, then update?  or wait for fix here.

Comment (bact) [reply docs/conformance.md:266]: If we can find it quickly today, maybe wait.
If not, we can have another PR to fix that.

**Pull request merged**